File: golden-paths/h1-foundation/python-fastapi/skeleton/src/app/routes/health.py
"""Health check routes."""

from datetime import datetime, timezone

from fastapi import APIRouter

router = APIRouter()


@router.get("/health")
async def health() -> dict[str, str]:
    """Basic health check endpoint.

    Returns the service status and current timestamp.
    """
    return {
        "status": "ok",
        "timestamp": datetime.now(tz=timezone.utc).isoformat(),
    }


@router.get("/ready")
async def ready() -> dict[str, bool]:
    """Readiness probe endpoint.

    Returns whether the service is ready to accept traffic.
    """
    return {
        "ready": True,
    }

File: golden-paths/h1-foundation/nodejs-microservice/skeleton/src/routes/health.ts
import { Router, Request, Response } from 'express';

export const healthRouter = Router();

/**
 * GET /health
 * Basic health check endpoint.
 * Returns the service status and current timestamp.
 */
healthRouter.get('/health', (_req: Request, res: Response) => {
  res.json({
    status: 'ok',
    timestamp: new Date().toISOString(),
  });
});

/**
 * GET /ready
 * Readiness probe endpoint.
 * Returns whether the service is ready to accept traffic.
 */
healthRouter.get('/ready', (_req: Request, res: Response) => {
  res.json({
    ready: true,
  });
});
